plugins/destination/mysql/cloud-config-ui/src/utils/formSchema.ts

```diff
@@ -1,88 +1,135 @@
+  const formFields = useMemo(() => {
+    const url = initialValues?.spec?.connection_string || '';
+    const connectionObj: Record<string, any> = convertConnectionStringToFields(url);
+    const partialSecretRegex = /\${[^}]+}/g;
 
-  // connection string settings
-  envs: yup
-    .array()
-    .of(
-      yup.object({
-        name: yup.string().default('').required(),
-        value: yup.string().default(''),
-      }),
-    )
-    .default([]),
-  connectionType: yup.string().oneOf(connectionTypeValues).default('fields').required(),
-  connectionString: yup
-    .string()
-    .default('')
-    .when('connectionType', {
-      is: 'string',
-      // eslint-disable-next-line unicorn/no-thenable
-      then: (schema) => schema.required(),
-    }),
-  username: yup.string().max(63).default(''),
-  password: yup.string().max(63).default(''),
-  host: yup.string().max(253).default(''),
-  port: yup
-    .string()
-    .max(5)
-    .matches(/^($)|(\d+)$/, 'Port must be a number')
-    .default(''),
-  database: yup.string().max(63).default(''),
-  tcp: yup.bool().default(true),
-  connectionParams: yup.object({
-    tls: yup.bool().default(false),
-    tlsMode: yup
-      .string()
-      .oneOf(tlsModeValues)
-      .default('preferred')
-      .when('tls', {
-        is: (tls: boolean) => !tls,
-        // eslint-disable-next-line unicorn/no-thenable
-        then: (schema: any) => schema.strip(),
-      }),
-    parseTime: yup.bool().default(false),
-    charset: yup.string().default(''),
-    loc: yup
-      .string()
-      .default('')
-      .when('parseTime', {
-        is: (parseTime: boolean) => !parseTime,
-        // eslint-disable-next-line unicorn/no-thenable
-        then: (schema: any) => schema.strip(),
-      }),
-    timeout: yup.number().integer().default(0),
-    readTimeout: yup.number().integer().default(0),
-    writeTimeout: yup.number().integer().default(0),
-    allowNativePasswords: yup.bool().default(false),
-  }),
+    return {
+      secretFields: {
+        password: yup
+          .string()
+          .max(63)
+          .default(connectionObj.password ? secretFieldValue : ''),
+      },
+      fields: {
+        connection_string: yup
+          .string()
+          .default(url.replaceAll(partialSecretRegex, secretFieldValue))
+          .when('_connectionType', {
+            is: 'string',
+            // eslint-disable-next-line unicorn/no-thenable
+            then: (schema) => schema.required(),
+          }),
+        username: yup
+          .string()
+          .max(63)
+          .default(connectionObj.username ?? ''),
+        host: yup
+          .string()
+          .max(253)
+          .default(connectionObj.host ?? ''),
+        port: yup
+          .string()
+          .max(5)
+          .matches(/^($)|(\d+)$/, 'Port must be a number')
+          .default(connectionObj.port ?? ''),
+        database: yup
+          .string()
+          .max(63)
+          .default(connectionObj.database ?? ''),
```

Comment:
None of those should be required?

Reply:
Per the go-sql-driver docs, it appears that a DNS string of `/` is valid, making all segments optional.
It is a little confusing in that it states both:
`Except for the databasename, all values are optional. So the minimal DSN is: /dbname`
and 
`If you do not want to preselect a database, leave dbname empty: /`
So the only value that is documented to be both required and optional is dbname 😂 

My take is that all can be optional and the test connection will be the source of truth if the connection gets made. Your thoughts?

ref: https://github.com/go-sql-driver/mysql?tab=readme-ov-file#dsn-data-source-name

Reply:
@jackcloudquery  - sorry, forgot to tag you

Reply:
Agreed